benches/common.sh

```diff
@@ -0,0 +1,109 @@
+# will not work if not under root
+# if benches are run localy by a developer droping caches might not be too important
+function maybe_drop_cache {
+	if test -w /proc/sys/vm/drop_caches; then
+		echo 3 > /proc/sys/vm/drop_caches
+	fi
+}
+
+function can_be_run_under_numa {
+	numactl "$@" true 1>/dev/null 2>/dev/null
+}
+
+# not every local machine has numa enabled on it
+# not every local machine has enough hardware units to run with intended configuration
+#
+# usage:
+# maybe_under_numactl numaoptions -- command to run
+# example:
+# maybe_under_numactl --membind=1 --cpunodebind=1 --physcpubind=11 -- echo qwe
+function maybe_under_numactl {
+	local numaoptions=()
+	local cmdoptions=()
+	local parsing_numa=1
+	for option in "$@"; do
+		if [ -n "$parsing_numa" ]; then
+			if [ "$option" == -- ]; then
+				parsing_numa=
+			else
+				numaoptions+=( "$option" )
+			fi
+		else
+			cmdoptions+=( "$option" )
+		fi
+	done
+
+	if which numactl 1>/dev/null 2>/dev/null; then
+		# check if it is even runable with given numactl options
+		if can_be_run_under_numa "${numaoptions[@]}"; then
+			numactl "${numaoptions[@]}" "${cmdoptions[@]}"
+		else
+			"${cmdoptions[@]}"
+		fi
+	else
+		"${cmdoptions[@]}"
+	fi
+}
+
+function get_tarantool_version {
+	"$TARANTOOL_EXECUTABLE" -v | grep -e "Tarantool" |  grep -oP '\s\K\S*'
+}
+
+function clean_tarantool {
+	rm -rf ./*.snap
+	rm -rf ./*.xlog
+	rm -rf ./*.vylog
+	rm -rf 5*
+}
+
+function sync_disk {
+	sync && echo "sync passed" || echo "sync failed with error: $?"
```

Comment:
Should not we do `echo 3 > /proc/sys/vm/drop_caches` as well? I know, it requires a root access, but maybe we can do something with it?

Reply:
I see, there is the `maybe_drop_cache` function. I was confused, because it is not used after `sync_disk` in `stop_and_clean_tarantool`.